### DISCUSSION TITLE
Как по кнопке во временной таблице перенести данные в расчетную?

### DISCUSSION THREAD

**Discussion by ToeraHL** [totumonline/totum-ru-issues-and-discussions#256] [Как это сделать?] [answered]
Здравствуйте!
Идея такая:
Данные в виде массива хранятся в ячейке расчетной таблице в цикле.
Временная таблица используется как интерфейс для чтения и изменения данных массива.
Что нужно написать в кнопке временной таблице, чтобы массив записался в ячейку строки, из которой была вызвана временная таблица? Просто через set не получается. Может быть надо передать id строки?
И еще вопрос: в созданном массиве будут храниться большие тексты, какой тип данных лучше использовать Данные или Текст?

Если такая логика будет сильно грузить сервер, то как лучше реализовать дополнительную вложенность данных внутри расчетной таблице в цикле (это получается следующий уровень)
Таблица циклов -> Расчетная таблица в цикле -> Еще таблица или массив

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Это достаточно часто используется. Для хранения массива используйте тип Данные. Если они у вас в строчной части, то при вызове временной таблицы в техническое поле в хедере передайте `id` строки и номер цикла (параметр `params` в `linkToDataTable`).

При `set` обратно вам надо указать куда этот `set` задав `cycle` и `where: 'id' = #h_param_id` (или как там вы назвали поле для сохранения `id` во временной табл).

По производительности проэксперементируйте с вашим максимальным объемом внутри цикла и если производительность ок — то ок. Дело в том, что когда таблица цикла загружается она читается из ячейки в БД целиком. И это чтение больших массивов будет замедлять любое действия.

**Если производительность не очень получается — то:**

1. Выносите все в отельную простую таблицу и делаете привязку к циклу: https://ru.docs.totum.online/tables#type-tab-in-cyclea

2. Для простой таблицы для поля данные можно задать`NOTLOADED` https://ru.docs.totum.online/fields#fields-settings-notLoaded — тогда оно будет читаться из БД только при прямом обращении к нему. Такая схема не будет нагружать сервер при работе с прочими элементами внутри цикла.
